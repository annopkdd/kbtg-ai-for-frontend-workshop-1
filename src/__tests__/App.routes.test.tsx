import { render } from '@testing-library/react';
import { MemoryRouter, Routes, Route } from 'react-router';
import Root from '../pages/Root';
import Home from '../pages/Home';
import TransactionDetail from '../pages/TransactionDetail';
import Transfer from '../pages/Transfer';
import History from '../pages/History';

// Mock the transaction data to ensure consistent snapshots
jest.mock('../data/transactions', () => ({
  getAllTransactions: jest.fn(() => [
    {
      id: 'TXN001',
      type: 'outgoing',
      recipient: 'สมชาย ใจดี',
      payTag: '@somchai123',
      amount: 500,
      memo: 'ค่าอาหารเที่ยง',
      date: 'วันนี้ 14:30',
      timestamp: new Date('2025-06-19T14:30:00'),
      status: 'completed',
      category: 'food',
      reference: 'REF001234567'
    },
    {
      id: 'TXN002',
      type: 'incoming',
      sender: 'นิดา สวยงาม',
      payTag: '@nida456',
      amount: 1200,
      memo: 'คืนเงินค่าหนัง',
      date: 'เมื่อวาน 19:45',
      timestamp: new Date('2025-06-18T19:45:00'),
      status: 'completed',
      category: 'entertainment',
      reference: 'REF001234568'
    }
  ]),
  getRecentTransactions: jest.fn(() => [
    {
      id: 'TXN001',
      type: 'outgoing',
      recipient: 'สมชาย ใจดี',
      payTag: '@somchai123',
      amount: 500,
      memo: 'ค่าอาหารเที่ยง',
      date: 'วันนี้ 14:30',
      timestamp: new Date('2025-06-19T14:30:00'),
      status: 'completed',
      category: 'food',
      reference: 'REF001234567'
    }
  ])
}));

describe('App Routes Snapshot Tests', () => {
  beforeEach(() => {
    // Mock Date.now to ensure consistent timestamps in snapshots
    jest.useFakeTimers();
    jest.setSystemTime(new Date('2025-06-19T10:00:00Z'));
  });

  afterEach(() => {
    jest.useRealTimers();
  });

  test('Root page (/) should match snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path="/" element={<Root />} />
        </Routes>
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Home page (/home) should match snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/home']}>
        <Routes>
          <Route path="/home" element={<Home />} />
        </Routes>
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Transfer page (/transfer) should match snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/transfer']}>
        <Routes>
          <Route path="/transfer" element={<Transfer />} />
        </Routes>
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('History page (/history) should match snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/history']}>
        <Routes>
          <Route path="/history" element={<History />} />
        </Routes>
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Transaction Detail page (/transaction/TXN001) should match snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/transaction/TXN001']}>
        <Routes>
          <Route path="/transaction/:id" element={<TransactionDetail />} />
        </Routes>
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  test('Transaction Detail page with invalid ID should match snapshot', () => {
    const { container } = render(
      <MemoryRouter initialEntries={['/transaction/INVALID_ID']}>
        <Routes>
          <Route path="/transaction/:id" element={<TransactionDetail />} />
        </Routes>
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});