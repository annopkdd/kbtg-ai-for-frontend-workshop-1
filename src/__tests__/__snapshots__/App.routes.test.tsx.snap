// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`App Routes Snapshot Tests History page (/history) should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center justify-between"
    >
      <div
        class="flex items-center space-x-3"
      >
        <a
          class="p-2 hover:bg-white/20 rounded-lg transition-colors"
          data-discover="true"
          href="/home"
        >
          <svg
            class="w-5 h-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 19l-7-7 7-7"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </a>
        <div>
          <h1
            class="text-xl font-semibold"
          >
            ประวัติการทำธุรกรรม
          </h1>
          <p
            class="text-web-green-100 text-sm"
          >
            0
             รายการ
          </p>
        </div>
      </div>
      <button
        class="p-2 hover:bg-white/20 rounded-lg transition-colors"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M3 4a1 1 0 011-1h16a1 1 0 011 1v2.586a1 1 0 01-.293.707l-6.414 6.414a1 1 0 00-.293.707V17l-4 4v-6.586a1 1 0 00-.293-.707L3.293 7.414A1 1 0 013 6.707V4z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </button>
    </div>
    <div
      class="mt-4"
    >
      <div
        class="relative"
      >
        <input
          class="w-full px-4 py-3 pl-10 bg-white/10 border border-white/20 rounded-lg text-white placeholder-white/70 focus:outline-none focus:ring-2 focus:ring-white/30"
          placeholder="ค้นหาตามชื่อ, PayTag, หรือหมายเหตุ..."
          type="text"
          value=""
        />
        <svg
          class="absolute left-3 top-3.5 w-4 h-4 text-white/70"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M21 21l-6-6m2-5a7 7 0 11-14 0 7 7 0 0114 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
    </div>
  </div>
  <div
    class="p-4"
  >
    <div
      class="flex items-center justify-center p-8"
    >
      <svg
        class="animate-spin h-8 w-8 text-web-green-600"
        fill="none"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          class="opacity-25"
          cx="12"
          cy="12"
          r="10"
          stroke="currentColor"
          stroke-width="4"
        />
        <path
          class="opacity-75"
          d="M4 12a8 8 0 018-8V0C5.373 0 0 5.373 0 12h4zm2 5.291A7.962 7.962 0 014 12H0c0 3.042 1.135 5.824 3 7.938l3-2.647z"
          fill="currentColor"
        />
      </svg>
    </div>
  </div>
</div>
`;

exports[`App Routes Snapshot Tests Home page (/home) should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center justify-between mb-4"
    >
      <div
        class="flex items-center space-x-3"
      >
        <div
          class="w-10 h-10 bg-white/20 rounded-full flex items-center justify-center"
        >
          <svg
            class="w-5 h-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 18h.01M8 21h8a2 2 0 002-2V5a2 2 0 00-2-2H8a2 2 0 00-2 2v14a2 2 0 002 2z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <div>
          <h1
            class="font-semibold"
          >
            สวัสดี, คุณสมศักดิ์
          </h1>
          <p
            class="text-web-green-100 text-sm"
          >
            ยินดีต้อนรับสู่ PayWise
          </p>
        </div>
      </div>
      <div
        class="flex space-x-2"
      >
        <button
          class="p-2 text-white hover:bg-white/20 rounded-lg transition-colors"
        >
          <svg
            class="w-5 h-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M15 17h5l-5-5 5-5h-5m-6 0H4l5 5-5 5h5"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
        <button
          class="p-2 text-white hover:bg-white/20 rounded-lg transition-colors"
        >
          <svg
            class="w-5 h-5"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M10.325 4.317c.426-1.756 2.924-1.756 3.35 0a1.724 1.724 0 002.573 1.066c1.543-.94 3.31.826 2.37 2.37a1.724 1.724 0 001.065 2.572c1.756.426 1.756 2.924 0 3.35a1.724 1.724 0 00-1.066 2.573c.94 1.543-.826 3.31-2.37 2.37a1.724 1.724 0 00-2.572 1.065c-.426 1.756-2.924 1.756-3.35 0a1.724 1.724 0 00-2.573-1.066c-1.543.94-3.31-.826-2.37-2.37a1.724 1.724 0 00-1.065-2.572c-1.756-.426-1.756-2.924 0-3.35a1.724 1.724 0 001.066-2.573c-.94-1.543.826-3.31 2.37-2.37.996.608 2.296.07 2.572-1.065z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
            <path
              d="M15 12a3 3 0 11-6 0 3 3 0 016 0z"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="bg-white/10 border border-white/20 text-white rounded-lg p-4"
    >
      <div
        class="flex items-center justify-between"
      >
        <div>
          <p
            class="text-web-green-100 text-sm"
          >
            ยอดเงินคงเหลือ
          </p>
          <div
            class="flex items-center space-x-2"
          >
            <p
              class="text-2xl font-bold"
            >
              ฿25,847.50
            </p>
            <button
              class="p-1 text-white hover:bg-white/20 rounded transition-colors"
            >
              <svg
                class="w-4 h-4"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M13.875 18.825A10.05 10.05 0 0112 19c-4.478 0-8.268-2.943-9.543-7a9.97 9.97 0 011.563-3.029m5.858.908a3 3 0 114.243 4.243M9.878 9.878l4.242 4.242M9.878 9.878L3 3m6.878 6.878L21 21"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="text-right"
        >
          <p
            class="text-web-green-100 text-sm"
          >
            บัญชีออมทรัพย์
          </p>
          <p
            class="text-sm"
          >
            XXX-X-X1234-X
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    class="p-4 space-y-4"
  >
    <div
      class="grid grid-cols-2 gap-4"
    >
      <a
        data-discover="true"
        href="/transfer"
      >
        <div
          class="bg-white border-2 border-neutral-200 hover:border-web-green-200 hover:shadow-md transition-all cursor-pointer rounded-lg p-4 text-center"
        >
          <div
            class="w-12 h-12 bg-web-green-100 rounded-full flex items-center justify-center mx-auto mb-2"
          >
            <svg
              class="w-6 h-6 text-web-green-600"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 19l9 2-9-18-9 18 9-2zm0 0v-8"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <h3
            class="font-semibold text-neutral-900"
          >
            โอนเงิน
          </h3>
          <p
            class="text-sm text-neutral-600"
          >
            PayWise Transfer
          </p>
        </div>
      </a>
      <a
        data-discover="true"
        href="/history"
      >
        <div
          class="bg-white border-2 border-neutral-200 hover:border-success hover:shadow-md transition-all cursor-pointer rounded-lg p-4 text-center"
        >
          <div
            class="w-12 h-12 bg-success/10 rounded-full flex items-center justify-center mx-auto mb-2"
          >
            <svg
              class="w-6 h-6 text-success"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 8v4l3 3m6-3a9 9 0 11-18 0 9 9 0 0118 0z"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
          </div>
          <h3
            class="font-semibold text-neutral-900"
          >
            ประวัติ
          </h3>
          <p
            class="text-sm text-neutral-600"
          >
            รายการทำธุรกรรม
          </p>
        </div>
      </a>
    </div>
    <div
      class="bg-white border border-neutral-200 rounded-lg shadow"
    >
      <div
        class="p-4 border-b border-neutral-200"
      >
        <div
          class="flex items-center justify-between"
        >
          <h2
            class="text-lg font-semibold text-neutral-900"
          >
            รายการล่าสุด
          </h2>
          <a
            data-discover="true"
            href="/history"
          >
            <button
              class="text-web-green-600 hover:text-web-green-700 text-sm font-medium"
            >
              ดูทั้งหมด
            </button>
          </a>
        </div>
      </div>
      <div
        class="p-4 space-y-3"
      >
        <a
          data-discover="true"
          href="/transaction/TXN001"
        >
          <div
            class="flex items-center justify-between p-3 rounded-lg hover:bg-neutral-50 cursor-pointer transition-colors"
          >
            <div
              class="flex items-center space-x-3"
            >
              <div
                class="w-10 h-10 rounded-full flex items-center justify-center bg-error/10"
              >
                <svg
                  class="w-5 h-5 text-error"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 11l5-5m0 0l5 5m-5-5v12"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div>
                <p
                  class="font-medium text-neutral-900"
                >
                  สมชาย ใจดี
                </p>
                <p
                  class="text-sm text-neutral-600"
                >
                  @somchai123
                </p>
                <p
                  class="text-xs text-neutral-500"
                >
                  ค่าอาหารเที่ยง
                </p>
              </div>
            </div>
            <div
              class="text-right"
            >
              <p
                class="font-semibold text-error"
              >
                -
                ฿
                500
              </p>
              <p
                class="text-xs text-neutral-500"
              >
                วันนี้ 14:30
              </p>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Routes Snapshot Tests Root page (/) should match snapshot 1`] = `
<div
  class="min-h-screen bg-gradient-to-br from-neutral-900 via-neutral-800 to-neutral-900 relative overflow-hidden flex items-center justify-center"
>
  <div
    class="absolute inset-0"
  >
    <div
      class="absolute top-1/4 left-1/4 w-96 h-96 bg-web-green-500/10 rounded-full blur-3xl animate-pulse"
    />
    <div
      class="absolute bottom-1/4 right-1/4 w-80 h-80 bg-web-green-400/15 rounded-full blur-3xl animate-pulse delay-1000"
    />
    <div
      class="absolute top-3/4 left-1/2 w-64 h-64 bg-web-green-300/10 rounded-full blur-3xl animate-pulse delay-500"
    />
  </div>
  <div
    class="relative z-10 text-center max-w-4xl mx-auto p-8"
  >
    <div
      class="bg-white/5 backdrop-blur-lg border border-white/10 rounded-3xl p-12 shadow-lg"
    >
      <div
        class="inline-flex items-center justify-center w-32 h-32 bg-gradient-to-r from-web-green-500 to-web-green-400 rounded-full mb-8 shadow-lg"
      >
        <svg
          class="w-16 h-16 text-white"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 8c-1.657 0-3 .895-3 2s1.343 2 3 2 3 .895 3 2-1.343 2-3 2m0-8c1.11 0 2.08.402 2.599 1M12 8V7m0 1v8m0 0v1m0-1c-1.11 0-2.08-.402-2.599-1M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </div>
      <h1
        class="text-7xl md:text-8xl font-bold bg-gradient-to-r from-white via-web-green-100 to-web-green-200 bg-clip-text text-transparent mb-6"
      >
        PayWise
      </h1>
      <div
        class="w-24 h-1 bg-gradient-to-r from-web-green-500 to-web-green-400 rounded-full mx-auto mb-8"
      />
      <p
        class="text-2xl md:text-3xl text-white/80 font-light leading-relaxed mb-12"
      >
        ระบบโอนเงินที่ทันสมัย
      </p>
      <div
        class="flex justify-center mb-8"
      >
        <a
          class="group bg-gradient-to-r from-web-green-600 to-web-green-500 hover:from-web-green-700 hover:to-web-green-600 text-white font-semibold py-6 px-8 rounded-2xl transition-all duration-300 transform hover:scale-105 hover:shadow-lg"
          data-discover="true"
          href="/home"
        >
          <div
            class="flex flex-col items-center space-y-3"
          >
            <svg
              class="w-8 h-8 group-hover:scale-110 transition-transform"
              fill="none"
              stroke="currentColor"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 12l2-2m0 0l7-7 7 7M5 10v10a1 1 0 001 1h3m10-11l2 2m-2-2v10a1 1 0 01-1 1h-3m-6 0a1 1 0 001-1v-4a1 1 0 011-1h2a1 1 0 011 1v4a1 1 0 001 1m-6 0h6"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
              />
            </svg>
            <span
              class="text-lg"
            >
              เริ่มต้นใช้งาน
            </span>
            <span
              class="text-sm opacity-80"
            >
              PayWise Dashboard
            </span>
          </div>
        </a>
      </div>
      <div
        class="text-center pt-6 border-t border-white/10"
      >
        <p
          class="text-white/50 text-sm"
        >
          React + TypeScript + TailwindCSS v4 + React Router
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`App Routes Snapshot Tests Transaction Detail page (/transaction/TXN001) should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center space-x-3 mb-4"
    >
      <a
        class="p-2 hover:bg-white/20 rounded-lg transition-colors"
        data-discover="true"
        href="/home"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 19l-7-7 7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
      <h1
        class="text-xl font-semibold"
      >
        รายละเอียดธุรกรรม
      </h1>
    </div>
  </div>
  <div
    class="p-4 space-y-4"
  >
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <div
        class="text-center mb-6"
      >
        <div
          class="w-16 h-16 rounded-full flex items-center justify-center mx-auto mb-4 bg-error/10"
        >
          <svg
            class="w-8 h-8 text-error"
            fill="none"
            stroke="currentColor"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 11l5-5m0 0l5 5m-5-5v12"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
            />
          </svg>
        </div>
        <p
          class="text-3xl font-bold mb-2 text-error"
        >
          -
          ฿
          500
        </p>
        <p
          class="text-lg font-medium text-neutral-900 mb-1"
        >
          โอนเงินให้
           
          สมชาย ใจดี
        </p>
        <p
          class="text-sm text-neutral-600"
        >
          @somchai123
        </p>
        <div
          class="inline-flex items-center px-3 py-1 rounded-full text-sm font-medium mt-3 bg-success/10 text-success"
        >
          สำเร็จ
        </div>
      </div>
    </div>
    <div
      class="bg-white rounded-lg p-6 shadow"
    >
      <h3
        class="text-lg font-semibold text-neutral-900 mb-4"
      >
        รายละเอียด
      </h3>
      <div
        class="space-y-4"
      >
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมายเลขอ้างอิง
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            REF001234567
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            วันที่และเวลา
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            19 มิถุนายน 2568 เวลา 14:30:00
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            บันทึก
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ค่าอาหารเที่ยง
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            จำนวนเงิน
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            ฿
            500
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            ค่าธรรมเนียม
          </span>
          <span
            class="font-medium text-green-600"
          >
            ฟรี
          </span>
        </div>
        <div
          class="flex justify-between"
        >
          <span
            class="text-neutral-600"
          >
            หมวดหมู่
          </span>
          <span
            class="text-neutral-900 font-medium"
          >
            อาหาร
          </span>
        </div>
        <div
          class="border-t border-neutral-200 pt-4"
        >
          <div
            class="flex justify-between"
          >
            <span
              class="text-neutral-600"
            >
              บัญชีปลายทาง
            </span>
            <div
              class="text-right"
            >
              <p
                class="text-neutral-900 font-medium"
              >
                @somchai123
              </p>
              <p
                class="text-sm text-neutral-600"
              >
                PayWise Account
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="space-y-3"
    >
      <button
        class="w-full bg-web-green-500 hover:bg-web-green-600 text-white py-3 px-4 rounded-lg font-medium transition-colors"
      >
        แสดงใบเสร็จ
      </button>
      <div
        class="grid grid-cols-2 gap-3"
      >
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          แชร์
        </button>
        <button
          class="bg-white border border-neutral-200 hover:border-web-green-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
        >
          ดาวน์โหลด
        </button>
      </div>
      <button
        class="w-full bg-neutral-100 hover:bg-neutral-200 text-neutral-700 py-3 px-4 rounded-lg font-medium transition-colors"
      >
        โอนซ้ำ
      </button>
    </div>
  </div>
</div>
`;

exports[`App Routes Snapshot Tests Transaction Detail page with invalid ID should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50 flex items-center justify-center"
>
  <div
    class="text-center"
  >
    <h1
      class="text-2xl font-bold text-neutral-900 mb-4"
    >
      ไม่พบรายการธุรกรรม
    </h1>
    <a
      class="bg-web-green-500 hover:bg-web-green-600 text-white px-6 py-3 rounded-lg transition-colors"
      data-discover="true"
      href="/home"
    >
      กลับหน้าหลัก
    </a>
  </div>
</div>
`;

exports[`App Routes Snapshot Tests Transfer page (/transfer) should match snapshot 1`] = `
<div
  class="min-h-screen bg-neutral-50"
>
  <div
    class="bg-gradient-to-r from-web-green-600 to-web-green-500 text-white p-4"
  >
    <div
      class="flex items-center space-x-3"
    >
      <a
        class="p-2 hover:bg-white/20 rounded-lg transition-colors"
        data-discover="true"
        href="/home"
      >
        <svg
          class="w-5 h-5"
          fill="none"
          stroke="currentColor"
          viewBox="0 0 24 24"
        >
          <path
            d="M15 19l-7-7 7-7"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
          />
        </svg>
      </a>
      <div>
        <h1
          class="text-xl font-semibold"
        >
          โอนเงิน
        </h1>
        <p
          class="text-web-green-100 text-sm"
        >
          PayWise Transfer
        </p>
      </div>
    </div>
    <div
      class="mt-6 flex items-center justify-center space-x-4"
    >
      <div
        class="flex items-center"
      >
        <div
          class="w-8 h-8 rounded-full flex items-center justify-center text-sm font-semibold bg-white text-web-green-600"
        >
          1
        </div>
        <div
          class="w-8 h-0.5 mx-2 bg-web-green-700"
        />
      </div>
      <div
        class="flex items-center"
      >
        <div
          class="w-8 h-8 rounded-full flex items-center justify-center text-sm font-semibold bg-web-green-700 text-web-green-200"
        >
          2
        </div>
        <div
          class="w-8 h-0.5 mx-2 bg-web-green-700"
        />
      </div>
      <div
        class="flex items-center"
      >
        <div
          class="w-8 h-8 rounded-full flex items-center justify-center text-sm font-semibold bg-web-green-700 text-web-green-200"
        >
          3
        </div>
      </div>
    </div>
  </div>
  <div
    class="p-4"
  >
    <div
      class="space-y-4"
    >
      <div
        class="bg-white border border-neutral-200 rounded-lg shadow"
      >
        <div
          class="p-4 border-b border-neutral-200"
        >
          <h2
            class="text-lg font-semibold text-neutral-900"
          >
            เลือกผู้รับ
          </h2>
        </div>
        <div
          class="p-4 space-y-4"
        >
          <div>
            <label
              class="block text-sm font-medium text-neutral-700 mb-2"
            >
              PayTag หรือเบอร์โทร
            </label>
            <div
              class="relative"
            >
              <input
                class="w-full px-4 py-3 border border-neutral-300 rounded-lg focus:ring-2 focus:ring-web-green-500 focus:border-transparent"
                placeholder="@username หรือ 0812345678"
                type="text"
                value=""
              />
              <button
                class="absolute right-2 top-2 p-2 text-neutral-500 hover:text-web-green-600"
              >
                <svg
                  class="w-5 h-5"
                  fill="none"
                  stroke="currentColor"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4v1m6 11h2m-6 0h-2v4m0-11v3m0 0h.01M12 12h4.01M16 20h4M4 12h4m12 0h.01M5 8h2a1 1 0 001-1V5a1 1 0 00-1-1H5a1 1 0 00-1 1v2a1 1 0 001 1zm12 0h2a1 1 0 001-1V5a1 1 0 00-1-1h-2a1 1 0 00-1 1v2a1 1 0 001 1zM5 20h2a1 1 0 001-1v-2a1 1 0 00-1-1H5a1 1 0 00-1 1v2a1 1 0 001 1z"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                  />
                </svg>
              </button>
            </div>
          </div>
          <div>
            <h3
              class="text-sm font-medium text-neutral-700 mb-3"
            >
              ผู้รับล่าสุด
            </h3>
            <div
              class="grid grid-cols-2 gap-3"
            >
              <button
                class="p-3 rounded-lg border-2 transition-all text-left border-neutral-200 hover:border-web-green-300"
              >
                <div
                  class="flex items-center space-x-3"
                >
                  <div
                    class="w-10 h-10 bg-gradient-to-r from-blue-400 to-purple-500 rounded-full flex items-center justify-center"
                  >
                    <span
                      class="text-white font-semibold text-sm"
                    >
                      ส
                    </span>
                  </div>
                  <div
                    class="flex-1 min-w-0"
                  >
                    <p
                      class="font-medium text-neutral-900 truncate"
                    >
                      สมชาย ใจดี
                    </p>
                    <p
                      class="text-sm text-neutral-600 truncate"
                    >
                      @somchai123
                    </p>
                  </div>
                  <div
                    class="w-2 h-2 bg-web-green-500 rounded-full"
                  />
                </div>
              </button>
              <button
                class="p-3 rounded-lg border-2 transition-all text-left border-neutral-200 hover:border-web-green-300"
              >
                <div
                  class="flex items-center space-x-3"
                >
                  <div
                    class="w-10 h-10 bg-gradient-to-r from-blue-400 to-purple-500 rounded-full flex items-center justify-center"
                  >
                    <span
                      class="text-white font-semibold text-sm"
                    >
                      น
                    </span>
                  </div>
                  <div
                    class="flex-1 min-w-0"
                  >
                    <p
                      class="font-medium text-neutral-900 truncate"
                    >
                      นิดา สวยงาม
                    </p>
                    <p
                      class="text-sm text-neutral-600 truncate"
                    >
                      @nida456
                    </p>
                  </div>
                  <div
                    class="w-2 h-2 bg-web-green-500 rounded-full"
                  />
                </div>
              </button>
              <button
                class="p-3 rounded-lg border-2 transition-all text-left border-neutral-200 hover:border-web-green-300"
              >
                <div
                  class="flex items-center space-x-3"
                >
                  <div
                    class="w-10 h-10 bg-gradient-to-r from-blue-400 to-purple-500 rounded-full flex items-center justify-center"
                  >
                    <span
                      class="text-white font-semibold text-sm"
                    >
                      ป
                    </span>
                  </div>
                  <div
                    class="flex-1 min-w-0"
                  >
                    <p
                      class="font-medium text-neutral-900 truncate"
                    >
                      ประยุทธ์ มั่นคง
                    </p>
                    <p
                      class="text-sm text-neutral-600 truncate"
                    >
                      @prayuth789
                    </p>
                  </div>
                </div>
              </button>
              <button
                class="p-3 rounded-lg border-2 transition-all text-left border-neutral-200 hover:border-web-green-300"
              >
                <div
                  class="flex items-center space-x-3"
                >
                  <div
                    class="w-10 h-10 bg-gradient-to-r from-blue-400 to-purple-500 rounded-full flex items-center justify-center"
                  >
                    <span
                      class="text-white font-semibold text-sm"
                    >
                      ม
                    </span>
                  </div>
                  <div
                    class="flex-1 min-w-0"
                  >
                    <p
                      class="font-medium text-neutral-900 truncate"
                    >
                      มาลี ใจงาม
                    </p>
                    <p
                      class="text-sm text-neutral-600 truncate"
                    >
                      @malee999
                    </p>
                  </div>
                </div>
              </button>
            </div>
          </div>
          <button
            class="w-full bg-web-green-600 hover:bg-web-green-700 text-white font-semibold py-3 px-4 rounded-lg transition-colors"
          >
            ถัดไป
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
